### ISSUE TITLE
Document additional features/functionalities in exchange for money

### ISSUE THREAD

**Issue by AdamPS** [nextcloud/appstore#1277]
Before using an app, I would like to know any features/functionalities it has that are only available in exchange for money. Otherwise I can spend a significant time installing it, only to find that it doesn't suit my requirements. The scope of this issue is intended to include any difference whatsoever in behaviour in the paid version, including "nags" or "different configuration options".

For example AFAIK:

1. OnlyOffice has a Premium model, with a key feature (editing on mobile) only available to paid users, with a license fee of €149 (Jan 2024) for home use, €180/year business.
2. Collabora office has a much more flexible model. In addition to the paid version (enterprise), there are 2 free versions, one limited by "nags", the other limited by scalability (home use). There is evidence of an undocumented configuration option that appears to bypass these limitations.

## Original text
(incorrect use of language - but preserved because the discussion below references it)

There is already the "Ethical AI rating". I propose that we could have another similar rating for "freedom" / FLOSS, which would apply in the case of Apps that require external 3rd party code. There could be further text that provides details of the freedom restrictions and how to choose/configure any relevant settings.

For example, it could work a bit like this:

- Entirely FLOSS would be green.
- Free = "Scalability limit” could be a yellow. This encourages funding the project from enterprises (who are making money and would otherwise pay the likes of Google), which is probably the most desirable from a user perspective.
- Free = “Nag-ware” seems like an orange. This is targeting ordinary end users (who could get a free, but-non-private service from Google), but it’s still usable.
- Free = “Key feature limitation” would be a red. The free service is barely usable, it's basically paid software. Perhaps these should cease to be “featured” apps, or even removed from the app store.

Collabora would count as a yellow overall, as they offer a yellow option (although it's not well documented). If they only offered the orange option that would of course count as an orange.

**Comment by gerhardt-bergmann** [comment]
In the introduction of your proposition you are speaking about **freedom**, which means a rating system that will measure the degree of freedom the app allows for its users, but in the examples and details that you provide you are speaking about **price**, neamely about the fact that some apps are entirely gratis to use, while others can be used in conjunction with some external software, like the Collabora Online server or the ONLYOFFICE Docs server, that the user has to pay for if he wants certain features.

Please be aware that you are making a serious confusion here. FOSS, which means **free and open source software** is a type of software that has nothing to do with price. It's all about freedom. The price is not taken into account at all. A FOSS application can cost money and yet be entirely FOSS, because FOSS is about the freedom a user has when using that application, which is guaranteed by the license that accompanies the application. So, when hearing FOSS, or its longer version FLOSS (free/libre and open source software), you should think about the license terms that come with the software to guarantee certain basic freedoms, and not about the price. Once again, FOSS doesn't tell you anything about the price. Many FOSS applications are gratis, but there are also many that are not gratis, like certain WordPress themes, Drupal  modules, etc.

Technically, FOSS means software that qulifies both as [**free software**](https://en.wikipedia.org/wiki/The_Free_Software_Definition) and as **open source software**. Both terms refer basically to the same software which respects the user's freedom. It doesn't matter if it's gratis or not. There are very few licenses that are "free software" licenses and not "open source" licenses (like the "Cryptix General License") and very few licenses that are "open source" licenses and not "free software" licenses (like the "Reciprocal Public License 1.5"). So, by using the term **free and open source software** you actually mean software that respects the user's freedom, because it is published under a license that guarantees at least the following 4 basic freedoms: to use the software as you wish, to modify the software as you wish, to give to others copies of the unmodified software, and to give to others copies of the modified software.

Nextcloud as a whole is published under a free software license, namely the [GNU Affero General Public License v3.0 (AGPLv3)](https://github.com/nextcloud/server/blob/master/COPYING). So, Nextcloud is free and open source software. Both apps [Collabora Online - Built-in CODE Server](https://github.com/CollaboraOnline/richdocumentscode/blob/master/LICENSE) and [ONLYOFFICE](https://github.com/ONLYOFFICE/onlyoffice-nextcloud/blob/master/LICENSE) are free and open source software, since they are published under the Apache License v2.0. The same can be said about all the apps published on the Nextcloud app store: they are all 100% free and open source software, because the licensing requirements stated [here](https://docs.nextcloud.com/server/latest/developer_manual/app_publishing_maintenance/publishing.html) ensure that all the apps published on the Nextcloud app store are published **under AGPLv3+ or any compatible license**.

Collabora Online Development Edition (CODE) in fact can be used without nags, without scalability issues (even if they say that CODE is for home and small teams), and totally gratis, if you install and properly configure the [Debian package](https://www.collaboraoffice.com/repos/CollaboraOnline/CODE-debian11/) or [Ubuntu package](https://www.collaboraoffice.com/repos/CollaboraOnline/CODE-arm64-ubuntu1804/) on your server, alongside Nextcloud, then use the [Nextcloud Office](https://apps.nextcloud.com/apps/richdocuments) app to connect it to Nextcloud.

The fact that Collabora Ltd and Ascensio System SIA (ONLYOFFICE) decided to distribute some versions of their servers as paid software doesn't have anything to do with the FOSS denomination of the apps they published on the Nextcloud app store. It's legal and it's their decision.

By using the expression **Apps that require external 3rd party code** you probably meant apps that link Nextcloud to a server installed on the same physical machine, or on the machine of a third party (like the third-party-hosted Collabora Online server). This is again legal if the interaction between the Nextcloud app and the other server complies with the terms of Nextcloud's AGPLv3 license, which is a free software license.

In conclusion, a FOSS rating system doesn't make sense, since all the apps published on the Nextcloud app store are equally FOSS.

**Comment by gerhardt-bergmann** [comment]
> your conclusion entirely fails to match my reality as a user of the system.

It actually matches your reality. You live in the same reality as us all and in this reality all the apps published in the Nextcloud app store are **free and open source software** because they are published under licenses which make them **free and open source software** as defined above. What you want to say is that you don't like certain apps, and especially the practices of the companies that develop those apps (like Collabora Ltd and Ascensio System SIA (ONLYOFFICE)), which add built-in restrictions to some versions of their programs that can be used in conjunction with those Nextcloud apps. I don't like those practices either. I don't like any limitations and techniques to force the users to do something. However, all the Nextcloud apps published on the app store are still 100% **free and open source software**, since their licenses guarantee the mentioned freedoms. This is why a FOSS rating system is not feasible. Please make the distinction between the classification of an app as **free and open source software** from a legal perspective, and the fact that some of their features are annoying. Their licenses actually give you the right to remove the annoying features at any moment and then use the software as you wish, even for commercial purposes (this implies that you have the skills to do it, or the money to pay somebody to do it, which is not always the case, but the fact that you have the right to do it is of enormous importance). Try to change the code of Windows, or of other (in)famous proprietary program. You obviously cannot because the proprietary software license doesn't even let you see the source code, let alone modify it and use it commercially.

On the [thread](https://help.nextcloud.com/t/office-alternatives-since-collabora-introduced-nagware-with-possibility-to-track-users/138236) that you mentioned, a user said that when you install CODE as a standalone server, without Docker (which is advisable, because when you install Docker you have to first debug the application installed inside it, and then debug Docker), you can permanently remove the welcome nagging window by adding the following lines in the `/etc/coolwsd/coolwsd.xml` file:
```
    <home_mode>
        <enable desc="Enable more configuration options for home users" type="bool" default="false">true</enable>
    </home_mode>
```
So, that thread shows that there is actually a very simple method to disable the welcome screen. I tried it on a standalone CODE server and it works.

ONLYOFFICE can be considered a type of "freemium", since it offers certain core features gratis, and offers additional functionalities for money. However, the term "freemium" doesn't say anything about the license terms and the actual freedoms granted or not to the users by the program's license. It only says something about the price: gratis for the core functionalities and paid for additional functionalities. Therefore the term "freemium" should be avoided and replaced with **free and open source software** vs **proprietary software**, which clearly speak about the fredom that the users are granted/not granted by the license, respectively.

You want a kind of system that will let you know from the start if an app published on the Nextcloud app store has restrictions or not, so that you won't waste time installinig it. This is fair enough. But this can't be **a FOSS rating system**, since all the apps are 100% FOSS as explained. The FOSS quality of an app is given by its license, and not by whether we like it or not. In the Nextcloud app store there is not one app that is less FOSS than the other, so as to deserve 4 stars instead of 5. What you really want is a section on each app's page, where the person who uploads the app explains if the app offers by itself or by connecting to an external server any features in exchange for money. This obviously implies that the app without those additional features has some shortcomings. This wouldn't be a **rating system**, but **the addition of a special section on each app's page, where it is mentioned if the app offers additional features/functionalities in exchange for money.** This could be done, but only the Nextcloud app store's admins can decide if they want to implement it or not.

**Comment by AdamPS** [comment]
> the addition of a special section on each app's page, where it is mentioned if the app offers additional features/functionalities in exchange for money

Yes you are exactly right. I updated the title.

> This could be done, but only the Nextcloud app store's admins can decide if they want to implement it or not.

Well of course, I am just making suggestions, and I hope that it goes without saying that the admins are free to decide as they wish.

**Comment by AdamPS** [comment]
I feel that the discussion about CODE belongs [here](https://github.com/nextcloud/documentation/issues/11433). Site owners would like to understand the Fremium/paid status. This applies for each of the available installation methods. I feel there is uncertainty over:

- scalability limits
- nags
- configuration options that might affect the above.

**Comment by gerhardt-bergmann** [comment]
Please remove the term **freemium** from the title of this issue. I mentioned in my previous post that the term freemium should be avoided because it speaks only about the price and not about the most important fact: the actual freedoms that the software license allows for users. In addition to this, it increases the confusion created by the word "free" related to software. We are part of the community that creates, maintains and promotes **free and open source software**. Therefore, we use the term **free** in relation to our software to signify **freedom**, not price. In **freemium** "free" signifies price (0 dollars for the core functionality), so, it doesn't speak about the actual freedoms that the users have when using the software.

We have to be very clear about what we stand for and about the language we use. If we mix up words, concepts and definitions, then our community is a joke and so is our software and our fight for freedom. Users' freedom is not an incidental quality of this type of software. It's the essence of it, so, we can't use the word **free** carelessly, or mix it up with other words in a way that creates confusion.

Also, your opening post is still problematic. You changed the title but you still speak about a "rating for "freedom" / FLOSS", when I explained that such a rating is unfeasible, since an app is either FOSS or not according to its license, and all the apps in the Nextcloud app store are FOSS, since this is a precondition for being admitted on the app store. You also speak about "freedom restrictions" referring to the freedom to do things, and then speak about "2 free versions" of Collabora, and "a free, but-non-private service from Google" where the term **free** means **free of charge**. When referring to price, the word **free** can be replaced with words that clearly signify price, like **gratis** or **costless**. Then you use the expression "Entirely FLOSS" in the sense that some apps are less FLOSS than others, which I explained it can't be said.

If you plan to reformulate the opening post, please notice the distinction between the Nextcloud apps like [ONLYOFFICE](https://apps.nextcloud.com/apps/onlyoffice) published on the Nextcloud app store, which are FOSS and don't impose restrictions upon the users, and the external programs that they link to, like the ONLYOFFICE Docs Community server, which are programs separate from Nextcloud that can be installed even on different physical machines, which do impose restrictions on the users, like [no more than 20 simultaneous connections](https://www.onlyoffice.com/compare-editions.aspx) for ONLYOFFICE Docs Community. So, there is a distinction between what is published on the Nextcloud app store, which has to be free and open source software to be admitted on the app store, and some external programs or "services", which those apps link to when configured to do so by the user. I must say I dislike ONLYOFFICE profoundly. I'm not defending them here. I'm just making clear the distinction between the 2 different components.

"Collabora Online - Built-in CODE Server" can't even be considered in this discussion. It's just a sort of demonstrative toy program. It's so slow that can't be used in real life, no matter how patient you might be. So, the only truly free and open source self-hosted alternative to Google Docs/Microsoft Office Online remains CODE installed as a Debian or Ubuntu package, like any other Linux package, or, in the worst case, installed using the Docker image.

**Comment by sunjam** [comment]
I'm not responding to the rating system discussion above, but isn't the underlying issue here a conceptual duplicate of
- #765 

As far I know the main confusion is that something else is needed to use a specific app, in this case running another service or paying for an external service.  Either way, confusion stems from an admin browsing through the app store and not understanding when installing a particular Nextcloud application is actually one step in a more advanced process necessary to setup that app into a usable state.

❤

**Comment by AdamPS** [comment]
@gerhardt-bergmann Apologies somehow I only just noticed your most recent reply. I have done my best to reformulate my opening post and adjust the title in the manner that you ask. However I don't have your expertise in this matter, so please forgive me if it is still incorrect.

@sunjam Thank you, that is a similar issue, but it isn't exactly the same. In this case, I was already aware of the external dependency, which was clearly explained. The paid feature was however a surprise.
